core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.html

```diff
@@ -37,7 +37,8 @@ <h4 tabindex="0" [ngClass]="(checkpointCount === 1) ? 'd-none' : 'progress-bar-l
             </div>
           </div>
           <div *ngFor="let checkpointNode of checkpointStatusArray; index as idx"
-               [ngClass]="{'completed-checkpoint-node': checkpointStatusArray[idx] === 'completed', 'in-progress-checkpoint-node': checkpointStatusArray[idx] === 'in-progress', 'incomplete-checkpoint-node': checkpointStatusArray[idx] === 'incomplete'}">
+               [ngClass]="{'completed-checkpoint-node': checkpointStatusArray[idx] === 'completed', 'in-progress-checkpoint-node': checkpointStatusArray[idx] === 'in-progress', 'incomplete-checkpoint-node': checkpointStatusArray[idx] === 'incomplete'}"
+               (click)="checkpointStatusArray[idx] === 'completed' ? returnToCheckpoint(idx) : null">
```

Comment:
Probably it's better to move this piece of logic into TS. Html is not a very good place for logic - it's harder to notice it there + it's difficult to test. The second part is important for your pr, you don't have tests that verify that clicking on an unfinished checkpoint doesn't move a user there, and that's a very important test.

I'd say rename "returnToCheckpoint" to "returnToCheckpointIfCompleted", call it here unconditionally, add an if to the method and add a test for it.

Reply:
Thank you. This is now being validated in "returnToCheckpointIfCompleted", in lesson-information-card-modal.component.ts